lib/index.js

```diff
@@ -977,6 +977,103 @@ class Mcdev {
+     * @param {string[]} [keys] customerkey of the metadata
+     * @returns {Array} array of fixedKeys
+     */
+    static async fixKeys(businessUnit, type, keys) {
+        Util.startLogger();
+        Util.logger.info('mcdev:: fixKeys');
+        const properties = await config.getProperties();
+        if (!(await config.checkProperties(properties))) {
+            // return null here to avoid seeing 2 error messages for the same issue
+            return null;
+        }
+        if (!type || !Util._isValidType(type)) {
+            return;
+        }
+
+        if (MetadataTypeDefinitions[type].keyField === MetadataTypeDefinitions[type].idField) {
+            Util.logger.error(`Key cannot be updated for this type`);
+            return null;
+        }
+        let [cred, bu] = businessUnit ? businessUnit.split('/') : [null, null];
+        let numItemsToUpdate;
+        let deployed;
+
+        const buObject = await Cli.getCredentialObject(
+            properties,
+            cred === null ? null : cred + '/' + bu,
+            null,
+            true
+        );
+        if (buObject !== null) {
+            cache.initCache(buObject);
+            cred = buObject.credential;
+            bu = buObject.businessUnit;
+        }
+        Util.logger.info(`\n               :: Updating keys for ${type} on ${cred}/${bu}\n`);
+        try {
+            try {
+                MetadataTypeInfo[type].client = auth.getSDK(buObject);
+            } catch (ex) {
+                Util.logger.error(ex.message);
+                return;
+            }
+            Util.logger.info(`First retrieve ${type}`);
+            const retriever = new Retriever(properties, buObject);
+            const retrieved = await retriever.retrieve([type], keys, null, false);
+
+            const typeArr = Object.values(retrieved)[0];
+            // merge results if multiple keys were provided
+            if (keys?.length > 1) {
+                const base = typeArr[0];
+                for (let i = 1; i < typeArr.length; i++) {
+                    // merge all items into the first array
+                    Object.assign(base, typeArr[i]);
+                }
+                // kick out all but first entry
+                typeArr.length = 1;
+            }
+            const keysForDeploy = MetadataTypeInfo[type].getKeysForFixing(typeArr[0], type);
+            numItemsToUpdate = keysForDeploy.length;
+            if (numItemsToUpdate < 1) {
+                Util.logger.info(`No items to update. Exiting`);
+                return true;
+            }
+            Util.OPTIONS.changeKeyField = MetadataTypeDefinitions[type].nameField;
+            properties.directories.deploy = properties.directories.retrieve;
+            deployed = await Deployer._deployBU(cred, bu, properties, [type], keysForDeploy, true);
+            const dependentTypes = await this.#getDependentMetadata(type);
+            await retriever.retrieve(dependentTypes, null, null, false);
```

Comment:
instead of retrieving all types here, lets instead print a warning to retrieve these these types. Otherwise we might be overwriting work that was prepared offline. It's might be unexpected to users that we would touch types that were not specified in the command.